### DISCUSSION TITLE
What is the meaning of `timeframe`?

### DISCUSSION THREAD

**Discussion by vvvlad** [jertel/elastalert2#485] [Q&A] [answered]
I've looked in documentation and was not able to find a specific reference to `timeframe` configuration.
The only reference was in relation with `scan_entire_timeframe`, but all examples of usage I see are without `scan_entire_timeframe`.
Would appreciate if someone could forward me to an explanation - what it is used for with `frequency` type for example.
How would this work with timeframe for `scan_entire_timeframe`?
thanks!

**Comment by ferozsalam** [comment, marked as answer]
There's a bit of detail [here](https://elastalert2.readthedocs.io/en/latest/running_elastalert.html?highlight=timeframe#creating-a-rule) -`timeframe` controls the period that ElastAlert looks back upon for a certain `num_events`. 

For for example, if you had a rule with the following:

```
run_every:
  minutes: 20
type: frequency
num_events: 5
timeframe:
  hours: 1
```

This would run the rule every twenty minutes, and each time that the rule was rule, ElastAlert would look back over the previous hour and count the number of occurences of your query. If the number was >5, it would alert.

Does this help?

**Reply by vvvlad** [reply]
Yes! this helps a lot! Thank you!

Some followups:
1. Would this mean that it will overlap same 20 minutes 3 times in your example?
2. what about the `scan_entire_timeframe`? if it also uses `timeframe` field, how would I use it both for `frequency` and `scan_entire_timeframe` at the same time?

**Reply by vvvlad** [reply]
Also, when searching for `timeframe` in the documentation, for some reason [this](https://elastalert2.readthedocs.io/en/latest/running_elastalert.html?highlight=timeframe#creating-a-rule) is not one of the results :(

**Reply by nsano-rururu** [reply]
`scan-entire-timeframe` in the documentation
https://elastalert.readthedocs.io/en/latest/ruletypes.html?highlight=scan_entire_timeframe#scan-entire-timeframe

There seems to be a bug in scan-entire-timeframe
[Spike alert not using specified timeframe for querying event counts](https://github.com/Yelp/elastalert/issues/3041)
[use rule timeframe when scan_entire_timeframe is set](https://github.com/Yelp/elastalert/pull/3141)
